ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -3551,24 +3551,123 @@ signal_list_t* create_operation_node(ast_node_t* op, signal_list_t** input_lists
     allocate_more_output_pins(operation_node, output_port_width);
     add_output_port_information(operation_node, output_port_width);
 
+    if ( (op->children[1]->type == NUMBERS) && (operation_node->type == SL || operation_node->type == SR || operation_node->type == ASR) ) {
```

Comment:
you should look for ASL also (although it is merged with SL, it improves readability)